frame/staking/src/pallet/impls.rs

```diff
@@ -662,6 +668,7 @@ impl<T: Config> Pallet<T> {
 		Self::do_remove_nominator(stash);
 
 		frame_system::Pallet::<T>::dec_consumers(stash);
+		T::EventListeners::on_unstake(stash);
```

Comment:
is it a bug in the existing system that we don't remove anyone here? I am surprised to see no `T::VoterList::on_remvoe` here.

Reply:
It's not a bug, afair it it's called in two places that already do a remove.

Reply:
If you look at this method you'll see that https://github.com/paritytech/substrate/blob/c04965378531a23fc7dfa0e39a71bc3cc9216126/frame/staking/src/pallet/impls.rs#L667-L668
`do_remove_validator` and `do_remove_nominator` are called just before `on_unstake`. Those methods call what you are looking for, namely `on_validator_remove` and `on_nominator_remove`.